flow/connectors/kafka/qrep.go

```diff
@@ -0,0 +1,111 @@
+package connkafka
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	lua "github.com/yuin/gopher-lua"
+
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+	"github.com/PeerDB-io/peer-flow/model"
+	"github.com/PeerDB-io/peer-flow/pua"
+)
+
+func (*KafkaConnector) SetupQRepMetadataTables(_ context.Context, _ *protos.QRepConfig) error {
+	return nil
+}
+
+func (c *KafkaConnector) SyncQRepRecords(
+	ctx context.Context,
+	config *protos.QRepConfig,
+	partition *protos.QRepPartition,
+	stream *model.QRecordStream,
+) (int, error) {
+	startTime := time.Now()
+	numRecords := atomic.Int64{}
+	schema := stream.Schema()
+
+	wgCtx, wgErr := context.WithCancelCause(ctx)
+	pool, err := c.createPool(wgCtx, config.Script, config.FlowJobName, wgErr)
+	if err != nil {
+		return 0, err
+	}
+	defer pool.Close()
+
+Loop:
+	for {
+		select {
+		case qrecord, ok := <-stream.Records:
+			if !ok {
+				c.logger.Info("flushing batches because no more records")
+				break Loop
+			}
+
+			pool.Run(func(ls *lua.LState) []*kgo.Record {
+				items := model.NewRecordItems(len(qrecord))
+				for i, val := range qrecord {
+					items.AddColumn(schema.Fields[i].Name, val)
+				}
+				record := &model.InsertRecord[model.RecordItems]{
+					BaseRecord:           model.BaseRecord{},
+					Items:                items,
+					SourceTableName:      config.WatermarkTable,
+					DestinationTableName: config.DestinationTableIdentifier,
+					CommitID:             0,
+				}
+
+				lfn := ls.Env.RawGetString("onRecord")
```

Comment:
doesn't cdc have to do something similar as well?

Reply:
Yes, but code is mixed up with tracking LSN/counts